pkg/frontend/adminactions/adminactions.go

```diff
@@ -0,0 +1,119 @@
+
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/dynamichelper"
+	"github.com/Azure/ARO-RP/pkg/env"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/compute"
+	"github.com/Azure/ARO-RP/pkg/util/refreshable"
+	"github.com/Azure/ARO-RP/pkg/util/restconfig"
+)
+
+// Interface for adminactions
+type Interface interface {
+	KubeGet(oc *api.OpenShiftCluster, groupKind, namespace, name string) ([]byte, error)
+	KubeList(oc *api.OpenShiftCluster, groupKind, namespace string) ([]byte, error)
+	KubeCreateOrUpdate(oc *api.OpenShiftCluster, obj *unstructured.Unstructured) error
+	KubeDelete(oc *api.OpenShiftCluster, groupKind, namespace, name string) error
+	MustGather(ctx context.Context, oc *api.OpenShiftCluster, w io.Writer) error
+	ResourcesList(ctx context.Context, fpAuth refreshable.Authorizer, subscriptionID, clusterRG, vNetID string) ([]byte, error)
+	SerialTool(ctx context.Context, w http.ResponseWriter, log *logrus.Entry,
+		fpAuth refreshable.Authorizer, r azure.Resource, oc *api.OpenShiftCluster, vmName string) error
+	VMRedeployAndWait(ctx context.Context, fpAuth refreshable.Authorizer, subscriptionID, clusterRG, vmName string) error
+}
+
+type adminactions struct {
+	log *logrus.Entry
+	env env.Interface
+}
+
+// New returns an adminactions Interface
+func New(log *logrus.Entry, env env.Interface) Interface {
+	return &adminactions{
+		log: log,
+		env: env,
+	}
+}
+
+func (a *adminactions) KubeGet(oc *api.OpenShiftCluster, groupKind, namespace, name string) ([]byte, error) {
+	restConfig, err := restconfig.RestConfig(a.env, oc)
+	if err != nil {
+		return nil, err
+	}
+	dh, err := dynamichelper.New(a.log, restConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	un, err := dh.Get(groupKind, namespace, name)
+	if err != nil {
+		return nil, err
+	}
+
+	return un.MarshalJSON()
+}
+
+func (a *adminactions) KubeList(oc *api.OpenShiftCluster, groupKind, namespace string) ([]byte, error) {
+	restConfig, err := restconfig.RestConfig(a.env, oc)
+	if err != nil {
+		return nil, err
+	}
+	dh, err := dynamichelper.New(a.log, restConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	ul, err := dh.List(groupKind, namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	return ul.MarshalJSON()
+}
+
+func (a *adminactions) KubeCreateOrUpdate(oc *api.OpenShiftCluster, obj *unstructured.Unstructured) error {
+	// TODO log changes
+
+	restConfig, err := restconfig.RestConfig(a.env, oc)
+	if err != nil {
+		return err
+	}
+	dh, err := dynamichelper.New(a.log, restConfig)
+	if err != nil {
+		return err
+	}
+
+	return dh.CreateOrUpdate(obj)
+}
+
+func (a *adminactions) KubeDelete(oc *api.OpenShiftCluster, groupKind, namespace, name string) error {
+	// TODO log changes
+
+	restConfig, err := restconfig.RestConfig(a.env, oc)
+	if err != nil {
+		return err
+	}
+	dh, err := dynamichelper.New(a.log, restConfig)
```

Comment:
see: https://msazure.visualstudio.com/AzureRedHatOpenShift/_backlogs/backlog/Azure%20Red%20Hat%20OpenShift/Features/?showParents=true&workitem=7862789 (this should go into the adminactions struct as well - could be a different PR)